src/private/quick/QWidgetAdapter_quick.cpp

```diff
@@ -300,23 +351,12 @@ QRect QWidgetAdapter::geometry() const
 
 QRect QWidgetAdapter::normalGeometry() const
 {
-    // TODO: There's no such concept in QWindow, do we need to workaround for QtQuick ?
-    return QWidgetAdapter::geometry();
+    return m_normalGeometry;
 }
 
 void QWidgetAdapter::setNormalGeometry(QRect geo)
```

Comment:
can you preserve the old code but just add m_normalGeometry = geo instead of the qDebug() ? Otherwise it's inconsistent with QWidgetAdapter_widgets.cpp's impl of setNormalGeometry()

And from an API point of view, the user of setNormalGeometry() would expect it to resize the window if the window as already normal
Also see  , which has a similar impl

Reply:
It is better not to set geometry here, because the method is called for every change in the position and size of the window.

I think setting the normal geometry is a special case and I would be surprised if the current geometry changes. (the normal geometry is in the private part of widgets and is missing for qml)

So,
setNormalGeometry is only used when restoring the size of the FloatingWindow.
But I noticed that before restoring to normal size (see FloatingWindow::deserialize) the deserializeWindowGeometry method is called. Geometry is restored twice.

Please, see my last commit